File: p76/p76.m
clear;
N = 100;
elems = 1:(N-1);
%coins = [1 2];
ways = zeros(1, N + 1);
ways(1) = 1;
for i = 1:length(elems)
    for j = elems(i):N
        ways(j + 1) = ways(j + 1) + ways(j + 1 - elems(i));
        %ways
    end
end
fprintf('number of ways is %.0f\n', ways(end));

% finally it's finished

File: p76/p76.cpp
#include <iostream>

using namespace std;

int main () {
	int target = 100;
	int ways[target + 1];;
	ways[0] = 1;
	 
	for (int i = 1; i <= 99; i++) {
	    for (int j = i; j <= target; j++) {
		ways[j] += ways[j - i];
	    }
	}
	cout << ways[99] << endl;
}